pyccel/codegen/printing/fcode.py

```diff
@@ -1819,6 +1819,10 @@ def _print_FunctionDef(self, expr):
             return ''
         self.set_scope(expr.scope)
 
+        for n in expr.results:
+            if isinstance(n, Variable) and n.is_ndarray:
+                raise NotImplementedError("Cannot convert a function ({}) that returns an array".format(expr.name))
```

Comment:
I think this should be in `_print_BindCFunctionDef`, otherwise you won't be able to add tests in `tests/pyccel`